src/components/EResourceFilters/EResourceFilters.js

```diff
@@ -1,52 +1,48 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
 
 import { Accordion, AccordionSet, FilterAccordionHeader, Selection } from '@folio/stripes/components';
-import { CheckboxFilter } from '@folio/stripes/smart-components';
+import { CheckboxFilter, MultiSelectionFilter } from '@folio/stripes/smart-components';
+
+const propTypes = {
+  activeFilters: PropTypes.object,
+  data: PropTypes.object.isRequired,
+  filterHandlers: PropTypes.object,
+};
 
 const FILTERS = [
   'publicationType',
   'type',
+  'tags'
 ];
 
-export default class EResourceFilters extends React.Component {
-  static propTypes = {
-    activeFilters: PropTypes.object,
-    data: PropTypes.object.isRequired,
-    filterHandlers: PropTypes.object,
-  };
-
-  static defaultProps = {
-    activeFilters: {
-      class: [],
-      publicationType: [],
-      type: [],
-    }
-  };
-
-  state = {
+export default function EResourceFilters({ activeFilters, data, filterHandlers }) {
+  const [filterState, setFilterState] = useState({
     publicationType: [],
     type: [],
-  }
+    tags: []
+  });
 
-  static getDerivedStateFromProps(props, state) {
+  useEffect(() => {
     const newState = {};
-
     FILTERS.forEach(filter => {
-      const values = props.data[`${filter}Values`] || [];
-      if (values.length !== state[filter].length) {
-        newState[filter] = values.map(({ label }) => ({ label, value: label }));
+      const values = data[`${filter}Values`];
+      if (values.length !== filterState[filter]?.length) {
+        newState[filter] = values;
       }
     });
 
-    if (Object.keys(newState).length) return newState;
+    if ((data?.tagsValues?.length ?? 0) !== filterState.tags?.length) {
+      newState.tags = data.tagsValues.map(({ label }) => ({ value: label, label }));
+    }
 
-    return null;
-  }
+    if (Object.keys(newState).length) {
+      setFilterState(prevState => ({ ...prevState, ...newState }));
+    }
+  }, [data, filterState]);
 
-  renderIsPackageFilter = () => {
-    const { activeFilters } = this.props;
+  const renderIsPackageFilter = () => {
```

Comment:
This looks functionally the same as renderCheckboxFilter, and is leading to a code smell. Could you try to get the package filters to use the generalised code please?